unstructured/ingest/connector/confluence.py

```diff
@@ -92,37 +94,88 @@ def filename(self):
+            "page_id": self.document_meta.document_id,
+        }
+
+    @requires_dependencies(["atlassian"], extras="Confluence")
+    def _get_page(self):
+        from atlassian import Confluence
+        from atlassian.errors import ApiError
+
+        try:
+            confluence = Confluence(
+                self.connector_config.url,
+                username=self.connector_config.user_email,
+                password=self.connector_config.api_token,
+            )
+            result = confluence.get_page_by_id(
+                page_id=self.document_meta.document_id,
+                expand="history.lastUpdated,version,body.view",
+            )
+        except ApiError as e:
+            logger.error(e)
+            return None
+        return result
+
+    def set_source_metadata(self, **kwargs):
+        """Fetches file metadata from the current page."""
+        page = kwargs.get("page", self._get_page())
+        if page is None:
+            self.source_metadata = SourceMetadata(
+                exists=False,
+            )
+        document_history = page["history"]
+        date_created = datetime.strptime(
+            document_history["createdDate"],
+            "%Y-%m-%dT%H:%M:%S.%fZ",
+        ).isoformat()
+        if last_updated := document_history.get("lastUpdated", "").get("when", ""):
+            date_modified = datetime.strptime(
+                last_updated,
+                "%Y-%m-%dT%H:%M:%S.%fZ",
+            ).isoformat()
+        else:
+            date_modified = date_created
+        version = page["version"]["number"]
+        self.source_metadata = SourceMetadata(
+            date_created=date_created,
+            date_modified=date_modified,
+            version=version,
+            source_url=page["_links"].get("self", None),
+            exists=True,
+        )
 
     @SourceConnectionError.wrap
     @requires_dependencies(["atlassian"], extras="confluence")
     @BaseIngestDoc.skip_if_file_exists
     def get_file(self):
-        from atlassian import Confluence
-
         logger.debug(f"Fetching {self} - PID: {os.getpid()}")
 
         # TODO: instead of having a separate connection object for each doc,
         # have a separate connection object for each process
-        confluence = Confluence(
-            self.connector_config.url,
-            username=self.connector_config.user_email,
-            password=self.connector_config.api_token,
-        )
 
-        result = confluence.get_page_by_id(page_id=self.file_meta.document_id, expand="body.view")
+        result = self._get_page()
+        if result is None:
+            raise ValueError(f"Failed to retrieve page with ID {self.document_meta.document_id}")
+
         self.document = result["body"]["view"]["value"]
         self.filename.parent.mkdir(parents=True, exist_ok=True)
         with open(self.filename, "w", encoding="utf8") as f:
             f.write(self.document)
+        self.set_source_metadata(page=result)
```

Comment:
maybe just move this up directly after we get the result (since this should be as close as possible to the fetch of the doc/content itself)